src/column_system.c

```diff
@@ -763,7 +763,7 @@ ecs_entity_t ecs_new_col_system(
     system_data->base.kind = kind;
     system_data->base.cascade_by = 0;
     system_data->base.has_refs = false;
-    system_data->base.needs_tables = ecs_needs_tables(world, sig);
+    system_data->base.needs_tables = ecs_needs_tables(world, sig, id);
```

Comment:
Suggestion: pass the system entity id (`result`) to the function. That way you can access all the information you need. You can get the id from a system like this:

```c
const char* id = ecs_get(world, result, EcsId);
```

Reply:
Oh, no you're not. Good point, looks like passing the system id is the right thing to do after all.